src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -229,6 +226,14 @@ private async Task ProcessQueueAsync()
             }
         }
 
+        private static async Task ExecuteCallbackAsync(QueueItem work, RequestContext context, CancellationToken queueCancellationToken)
+        {
+            // Create a linked cancellation token to cancel any requests in progress when this shuts down
+            using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(queueCancellationToken, work.CancellationToken);
+
+            await work.CallbackAsync(context, cancellationTokenSource.Token).ConfigureAwait(false);
```

Comment:
💡 The `CombineWith` extension only creates a new object when _both_ tokens can be cancelled, so it's preferred whenever the project references Microsoft.VisualStudio.Threading.

```suggestion
            // Create a combined cancellation token to cancel any requests in progress when this shuts down
            using var combinedTokenSource = queueCancellationToken.CombineWith(work.CancellationToken);

            await work.CallbackAsync(context, combinedTokenSource.Token).ConfigureAwait(false);
```